core/src/main/java/io/grpc/inprocess/InProcessChannelBuilder.java

```diff
@@ -61,14 +62,22 @@ private InProcessChannelBuilder(String name) {
   }
 
   @Override
-  protected ChannelEssentials buildEssentials() {
-    final String name = this.name;
-    ClientTransportFactory transportFactory = new ClientTransportFactory() {
-      @Override
-      public ClientTransport newClientTransport() {
-        return new InProcessTransport(name);
-      }
-    };
-    return new ChannelEssentials(transportFactory, null);
+  protected ClientTransportFactory buildTransportFactory() {
+    return new InProcessClientTransportFactory();
+  }
+
+  private final class InProcessClientTransportFactory extends AbstractReferenceCounted
+          implements ClientTransportFactory {
+    private final String name = InProcessChannelBuilder.this.name;
```

Comment:
Oh, I see it saves the current state of the builder. It would be _really nice_ for the class to be static, so that we can detect if a field is used without being saved.